logstash-core/lib/logstash/instrument/metric_store.rb

```diff
@@ -180,6 +180,16 @@ def each(path = nil, &block)
     end
     alias_method :all, :each
 
+    def prune(path)
+      key_paths = path.gsub(/^#{KEY_PATH_SEPARATOR}+/, "").split(KEY_PATH_SEPARATOR)
+      key_paths = key_paths.map {|k| k.to_sym }
```

Comment:
You can use the symbol to proc here.map(&:to_sym) this could be also part of the `#keys_path` method.